sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricsEnvironment.java

```diff
@@ -53,6 +54,12 @@
 
   private static final ThreadLocal<@Nullable MetricsContainer> CONTAINER_FOR_THREAD =
       new ThreadLocal<>();
+  private static final AtomicReference<MetricsContainer> CONTAINER_GLOBAL = new AtomicReference<>();
```

Comment:
As mentioned above, please exclude this change from the pr and discuss with @Zhangyx39